### PR TITLE
feature/wishlist

### PR DESCRIPTION
**Pull request by DilnazTolubaiKyzy**

### CODE REVIEW

**Review by Aidai0404 — COMMENTED** [review]
viewed

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:1]: Why you touching BasketController class?

File: src/main/java/us/peaksoft/gadgetarium/controller/ChosenController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        return chosenService.saveProductInChosen(id, productRequest);
+    }
+
+    @PutMapping("products/{id}")
+    @Operation(description = "Users and Admin can delete a Product from Wishlist")
+    public SimpleResponse deleteProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return chosenService.deleteProductFromChosen(id, productRequest);
+    }
+
+    @GetMapping("all-products/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/ChosenController.java:40]: Here id is which id?

File: src/main/java/us/peaksoft/gadgetarium/dto/BasketRequest.java
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/BasketRequest.java:1]: Why you touching Basket?

File: src/main/java/us/peaksoft/gadgetarium/dto/BasketResponse.java
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/BasketResponse.java:1]: Why you touching Basket?

File: src/main/java/us/peaksoft/gadgetarium/dto/ChosenRequest.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ChosenRequest {
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/ChosenRequest.java:12]: Rename to wishlist

File: src/main/java/us/peaksoft/gadgetarium/dto/ChosenResponse.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ChosenResponse {
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/ChosenResponse.java:12]: Rename to wishlist

File: src/main/java/us/peaksoft/gadgetarium/repository/ChosenRepository.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import us.peaksoft.gadgetarium.entity.Chosen;
+import us.peaksoft.gadgetarium.entity.Product;
+
+import java.util.List;
+
+
+@Repository
+@Transactional
+public interface ChosenRepository extends JpaRepository<Chosen, Long> {
+    @Query("SELECT pro FROM Product pro JOIN pro.chosen cho WHERE cho.id =:id")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/repository/ChosenRepository.java:18]: Next line

File: src/main/java/us/peaksoft/gadgetarium/service/AuthService.java
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/service/AuthService.java:1]: Why you touching other classes?

**Review by i-Coderr — COMMENTED** [review]
**Please don't touch other classes not belong to your task**

File: src/main/java/us/peaksoft/gadgetarium/controller/WishlistController.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import us.peaksoft.gadgetarium.dto.ProductRequest;
+import us.peaksoft.gadgetarium.dto.ProductResponse;
+import us.peaksoft.gadgetarium.dto.SimpleResponse;
+import us.peaksoft.gadgetarium.service.WishlistService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/wishlist")
+@Tag(name = "BasketController", description = "API endpoints for managing Wishlist")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/WishlistController.java:17]: What is this? Just copy and pass?

**Review by i-Coderr — APPROVED** [review]
Alright

**Pull request merged**